crypto/keyring/keyring.go

```diff
@@ -144,6 +144,15 @@ type Options struct {
 	SupportedAlgos SigningAlgoList
 	// supported signing algorithms for Ledger
 	SupportedAlgosLedger SigningAlgoList
+	// define Ledger Derivation function
+	LedgerDerivation func() (ledger.SECP256K1, error)
```

Comment:
Can't ledger derive keys that aren't on the SECP curve?

Reply:
Currently, Ledger devices are represented using the `SECP256k1` interface. This PR intended only to generalize Ledger support within the realm of secp256k1 keys, but it's possible that generalizing it even further might be more appropriate. Do you think that would be the case?